### PR TITLE
TST add test for exception rtti passing

### LINKED ISSUE #61: more OSX RTTI issues
Issue: I finally got this test case finished. Interestingly, it now fails completely instead of only partially like last time. 

The code is here: https://github.com/beckermr/misc/tree/master/work/pybind11_conda_exceptions
The readme has a basic description of how to run the test. 

On osx I get (with either the system or the conda compiler)

```
========================================================================== test session starts ===========================================================================
platform darwin -- Python 3.7.6, pytest-5.4.1, py-1.8.1, pluggy-0.13.0 -- /Users/beckermr/miniconda3/envs/pybind11-test/bin/python
cachedir: .pytest_cache
rootdir: /Users/beckermr/Desktop/misc/work/pybind11_conda_exceptions
collecting ... dyld: loaded: /Users/beckermr/Desktop/misc/work/pybind11_conda_exceptions/exception_test/exceptions.cpython-37m-darwin.so
dyld: loaded: /Users/beckermr/Desktop/misc/work/pybind11_conda_exceptions/tests/testLib.cpython-37m-darwin.so
collected 1 item                                                                                                                                                         

test_exception_catching_python.py::test_exception_catching_python_smoke 
==========
standard throw!
==========
FAILED

================================================================================ FAILURES ================================================================================
__________________________________________________________________ test_exception_catching_python_smoke __________________________________________________________________

    def test_exception_catching_python_smoke():
        with pytest.raises(exception_test.LSSTException):
>           testLib.fail("message1")
E           RuntimeError: message1

test_exception_catching_python.py:9: RuntimeError
======================================================================== short test summary info =========================================================================
FAILED test_exception_catching_python.py::test_exception_catching_python_smoke - RuntimeError: message1
=========================================================================== 1 failed in 0.11s ============================================================================
```

The runtime error is pybind11 doing its own thing with the uncaught exception in the exception translator block.

On linux, it works fine

```
========================================================================== test session starts ===========================================================================
platform linux -- Python 3.7.6, pytest-5.4.1, py-1.8.1, pluggy-0.12.0 -- /opt/conda/envs/pybind11-test/bin/python
cachedir: .pytest_cache
rootdir: /root/misc/work/pybind11_conda_exceptions
collected 1 item                                                                                                                                                         

test_exception_catching_python.py::test_exception_catching_python_smoke 
==========
standard throw!
==========

==========
caught
==========
PASSED

=========================================================================== 1 passed in 0.02s ============================================================================
/root/misc/work/pybind11_conda_exceptions
```

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
# packages in environment at /Users/beckermr/miniconda3/envs/pybind11-test:
#
# Name                    Version                   Build  Channel
appnope                   0.1.0           py37hc8dfbb8_1001    conda-forge
attrs                     19.3.0                     py_0    conda-forge
backcall                  0.1.0                      py_0    conda-forge
binutils-meta             1.0.4                         0    conda-forge
c-compiler                1.0.4                h1239861_0    conda-forge
ca-certificates           2020.4.5.1           hecc5488_0    conda-forge
cctools                   927.0.2              h5ba7a2e_4    conda-forge
certifi                   2020.4.5.1       py37hc8dfbb8_0    conda-forge
clang                     9.0.0           default_hf57f61e_4    conda-forge
clang_osx-64              9.0.0                h22b1bf0_3    conda-forge
clangxx                   9.0.0           default_hf57f61e_4    conda-forge
clangxx_osx-64            9.0.0                h05bbb7f_6    conda-forge
compiler-rt               9.0.0                hce3ea14_0    conda-forge/label/broken
compilers                 1.0.4                         0    conda-forge
cxx-compiler              1.0.4                h707564b_0    conda-forge
decorator                 4.4.2                      py_0    conda-forge
exception-test            0.1.0                     dev_0    <develop>
fortran-compiler          1.0.4                he991be0_0    conda-forge
gfortran_impl_osx-64      7.3.0                hf4212f2_2    conda-forge
gfortran_osx-64           7.3.0                h22b1bf0_9    conda-forge
gmp                       6.2.0                h4a8c4bd_2    conda-forge
importlib-metadata        1.6.0            py37hc8dfbb8_0    conda-forge
importlib_metadata        1.6.0                         0    conda-forge
ipython                   7.13.0           py37hc8dfbb8_2    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
isl                       0.19                          0    conda-forge
jedi                      0.17.0           py37hc8dfbb8_0    conda-forge
ld64                      450.3                h3c32e8a_3    conda-forge
libcxx                    9.0.1                         1    conda-forge
libffi                    3.2.1             h4a8c4bd_1007    conda-forge
libgfortran               4.0.0                         2    conda-forge
libiconv                  1.15              h0b31af3_1006    conda-forge
libllvm9                  9.0.1                h7475705_1    conda-forge
llvm-openmp               10.0.0               h28b9765_0    conda-forge
more-itertools            8.2.0                      py_0    conda-forge
mpc                       1.1.0             h4160ff4_1007    conda-forge
mpfr                      4.0.2                h65ac59c_1    conda-forge
ncurses                   6.1               h0a44026_1002    conda-forge
openssl                   1.1.1f               h0b31af3_0    conda-forge
packaging                 20.1                       py_0    conda-forge
parso                     0.7.0              pyh9f0ad1d_0    conda-forge
pexpect                   4.8.0            py37hc8dfbb8_1    conda-forge
pickleshare               0.7.5           py37hc8dfbb8_1001    conda-forge
pip                       20.0.2                     py_2    conda-forge
pluggy                    0.13.0                   py37_0    conda-forge
prompt-toolkit            3.0.5                      py_0    conda-forge
ptyprocess                0.6.0                   py_1001    conda-forge
py                        1.8.1                      py_0    conda-forge
pybind11                  2.5.0            py37ha1cc60f_0    conda-forge
pygments                  2.6.1                      py_0    conda-forge
pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
pytest                    5.4.1            py37hc8dfbb8_0    conda-forge
python                    3.7.6           h90870a6_5_cpython    conda-forge
python_abi                3.7                     1_cp37m    conda-forge
readline                  8.0                  hcfe32e1_0    conda-forge
setuptools                46.1.3           py37hc8dfbb8_0    conda-forge
six                       1.14.0                     py_1    conda-forge
sqlite                    3.30.1               h93121df_0    conda-forge
tapi                      1000.10.8            ha1b3eb9_4    conda-forge
tk                        8.6.10               hbbe82c9_0    conda-forge
traitlets                 4.3.3            py37hc8dfbb8_1    conda-forge
wcwidth                   0.1.9              pyh9f0ad1d_0    conda-forge
wheel                     0.34.2                     py_1    conda-forge
xz                        5.2.5                h0b31af3_0    conda-forge
zipp                      3.1.0                      py_0    conda-forge
zlib                      1.2.11            h0b31af3_1006    conda-forge
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

     active environment : pybind11-test
    active env location : /Users/beckermr/miniconda3/envs/pybind11-test
            shell level : 2
       user config file : /Users/beckermr/.condarc
 populated config files : /Users/beckermr/.condarc
          conda version : 4.8.3
    conda-build version : 3.19.2
         python version : 3.7.6.final.0
       virtual packages : __osx=10.14.6
       base environment : /Users/beckermr/miniconda3  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/beckermr/miniconda3/pkgs
                          /Users/beckermr/.conda/pkgs
       envs directories : /Users/beckermr/miniconda3/envs
                          /Users/beckermr/.conda/envs
               platform : osx-64
             user-agent : conda/4.8.3 requests/2.23.0 CPython/3.7.6 Darwin/18.7.0 OSX/10.14.6
                UID:GID : 501:20
             netrc file : /Users/beckermr/.netrc
           offline mode : False


```
</details>

cc @isuruf

### LINKED ISSUE #63: tests were not running and now fail
Due to conda build not running the tests (!!!), it appears that version 10 is somewhat broken. 

On linux, we cannot even solve for the test env

```
Output in format: Requested package -> Available versions

Package libstdcxx-ng conflicts for:
clang -> libstdcxx-ng[version='>=7.3.0']
gxx_linux-64=7 -> libstdcxx-ng==7.2.0=h24385c6_1

Package libcxxabi conflicts for:
libcxxabi==10.0.0=2
libcxx==10.0.0=hc9558a2_2 -> libcxxabi==10.0.0=2

```

on OSX we are hitting a variety of errors due to the linker not recgonizing `-platform-version` flags and if you try a hack to fix that, you get 

```
clang++ -stdlib=libc++ -O2 -g test_sources/test_exceptions.cpp -Wl,-rpath,/Users/travis/miniconda3/conda-bld/libcxx_and_libcxxabi_1587728061005/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib -L/Users/travis/miniconda3/conda-bld/libcxx_and_libcxxabi_1587728061005/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib -mlinker-version=0
Assertion failed: (!"Unknown operand"), function LinkLocation, file /SourceCache/dwarf_utilities/dwarf_utilities-121.1/source/DWARFdSYM.cpp, line 1818.
clang-10: error: unable to execute command: Abort trap: 6
clang-10: error: dsymutil command failed due to signal (use -v to see invocation)
```

cc @conda-forge/libcxx @isuruf

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds tests for exception passing on OSX.

closes #61 
closes #63 

Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
thanks @isuruf! I'm good to merge here if you are!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -73,6 +49,36 @@ outputs:
     requirements:   # [linux]
       run:   # [linux]
         - {{ pin_subpackage("libcxxabi", exact=True) }}    # [linux]
+    test:
+      requires:
+        # osx tests for exception passing fail on vlang v10 due to exception in clang
+        - clang <10              # [osx]
```
Comment (isuruf) [inline recipe/meta.yaml:55]: Do you have a log with clang 10?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -73,6 +49,36 @@ outputs:
...
+        - clang                  # [not osx]
+        - {{ compiler('cxx') }}  # [not osx]
```
Comment (isuruf) [inline recipe/meta.yaml:57]: Can you remove `not osx`?

**Comment by isuruf** [comment]
Btw, I think this should be a downstream test. I don't want to introduce a cyclic dependency involving python. The bot might go crazy.

**Comment by beckermr** [comment]
I agree with, but the package which has this code is not on conda/conda-forge. Should we make a special package just for this test?

**Comment by isuruf** [comment]
Or add this test to pybind11 and we can add pybind11 as a downstream test.

**Comment by beckermr** [comment]
Sounds good. Let's keep this code intact here until I can move it to the new repo. Then we can just push the fixes for the tests in this PR with pybind11 downstreamed.

File: recipe/meta.yaml (hunk 1 above)
Comment (beckermr) [reply recipe/meta.yaml]: Here is one: https://travis-ci.com/github/conda-forge/libcxx-feedstock/builds/161846808

File: recipe/meta.yaml (hunk 2 above)
Comment (beckermr) [reply recipe/meta.yaml]: Removing this caused a huge number of clobber warnings and then things failed:

https://travis-ci.com/github/conda-forge/libcxx-feedstock/builds/161848765

Comment (beckermr) [reply recipe/meta.yaml]: Also somehow it pulled in an old copy of libcxxabi. 

```
'[' -f /Users/travis/miniconda3/conda-bld/libcxx_and_libcxxabi_1587739468274/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/libc++abi.dylib ']'
```

Comment (beckermr) [reply recipe/meta.yaml]: What is clang bootstrap doing here?

```
ca-certificates:        2020.4.5.1-hecc5488_0     conda-forge
    certifi:                2020.4.5.1-py38h32f6830_0 conda-forge
    clang:                  9.0.1-default_hf57f61e_0  conda-forge
    clang-tools:            9.0.1-default_hf57f61e_0  conda-forge
    clang_bootstrap_osx-64: 9.0.0-h22b1bf0_3          conda-forge
    clangxx:                9.0.1-default_hf57f61e_0  conda-forge
    libcxx:                 10.0.0-hd34085c_2         local      
    libffi:                 3.2.1-h4a8c4bd_1007       conda-forge
    libllvm9:               9.0.1-h7475705_1          conda-forge
    make:                   4.3-h0b31af3_0            conda-forge
    ncurses:                6.1-h0a44026_1002         conda-forge
    openssl:                1.1.1g-h0b31af3_0         conda-forge
    pip:                    20.0.2-py_2               conda-forge
    pybind11:               2.5.0-py38ha0d09dd_0      conda-forge
    python:                 3.8.2-hd5f0129_7_cpython  conda-forge
    python_abi:             3.8-1_cp38                conda-forge
    readline:               8.0-hcfe32e1_0            conda-forge
    setuptools:             46.1.3-py38h32f6830_0     conda-forge
    sqlite:                 3.30.1-h93121df_0         conda-forge
    tk:                     8.6.10-hbbe82c9_0         conda-forge
    wheel:                  0.34.2-py_1               conda-forge
    xz:                     5.2.5-h0b31af3_0          conda-forge
    zlib:                   1.2.11-h0b31af3_1006      conda-forge
```

**Comment by beckermr** [comment]
Alright. Once we merge the pybind11 PR and the CDN sync is done, we can rerun this one to make sure the test runs ok and then merge it!

**Comment by beckermr** [comment]
@conda-forge-admin please restart ci

**Comment by beckermr** [comment]
OK @isuruf @conda-forge/libcxx this one is ready to merge!

**Pull request merged**